sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/NamespaceNode.scala

```diff
@@ -0,0 +1,39 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.sql.catalyst.analysis
+
+import org.apache.spark.sql.catalyst.expressions.Attribute
+import org.apache.spark.sql.catalyst.plans.logical.LeafNode
+import org.apache.spark.sql.connector.catalog.SupportsNamespaces
+
+trait NamespaceNode extends LeafNode {
+  override def output: Seq[Attribute] = Nil
+  def catalog: SupportsNamespaces
+  def namespace: Seq[String]
+}
+
+case class ResolvedNamespace(catalog: SupportsNamespaces, namespace: Seq[String])
+  extends NamespaceNode
+
+case class UnresolvedNamespace(multipartIdentifier: Seq[String]) extends NamespaceNode {
```

Comment:
BTW, we should give nice error message in `CheckAnalysis` when we hit `UnresolvedNamespace` and `UnresolvedV2Table`.